### PR TITLE
Add header mdn

### PR DESCRIPTION
**Pull request by sherlockvn**

### CODE REVIEW

File: test/jmap/email/set/set_email_method_test.dart [hunk 1]
```diff
@@ -261,5 +261,121 @@ void main() {
...
+                {
+                  "accountId":
+                      "587a9c5a4a9c0a4d36243b7417700d5383cbbfa25f0909ab7f6f4baaa5bfde9b",
+                  "create": {
+                    "e01": {
+                      "id": "e102",
+                      "mailboxIds": {
+                        "a6f488c0-964b-11ec-83d6-c1ded34233a9": true
+                      },
+                      "Subject": "[POSTMAN] SEND EMAIL WITH MDN MDN MDN",
```
Comment (dab246) [inline test/jmap/email/set/set_email_method_test.dart:312]: `Subject ` is lowercase

File: test/jmap/email/set/set_email_method_test.dart [hunk 2]
```diff
@@ -261,5 +261,121 @@ void main() {
...
+                      "From": [
```
Comment (dab246) [inline test/jmap/email/set/set_email_method_test.dart:313]: same

File: test/jmap/email/set/set_email_method_test.dart [hunk 3]
```diff
@@ -261,5 +261,121 @@ void main() {
...
+                      },
+                      "0"
+                  ]
+              ]
+          }),
+          data: {
+            "using": [
+              "urn:ietf:params:jmap:core",
+              "urn:ietf:params:jmap:mail",
+              "urn:apache:james:params:jmap:mail:shares"
```
Comment (dab246) [inline test/jmap/email/set/set_email_method_test.dart:298]: `SetEmailMethod` no need for this capability:  `urn:apache:james:params:jmap:mail:shares`

File: test/jmap/email/set/set_email_method_test.dart [hunk 4]
```diff
@@ -261,5 +261,121 @@ void main() {
...
+                          "value": "[POSTMAN] SEND EMAIL WITH MDN",
+                          "isEncodingProblem": false,
+                          "isTruncated": false
+                        }
+                      },
+                      "header:Disposition-Notification-To:asText": "qkdo@linagora.com"
+                    }
+                  }
+                },
+                "0"
```
Comment (dab246) [inline test/jmap/email/set/set_email_method_test.dart:333]: it always starts at `c0` if you don't set `methodId`

File: test/jmap/email/set/set_email_method_test.dart [hunk 5]
```diff
@@ -261,5 +261,121 @@ void main() {
...
+                          "created": {
+                              "e01": {
+                                  "id": "5e2b02d0-4ab0-11ed-88ee-ffc86e0cde67",
+                                  "blobId": "5e2b02d0-4ab0-11ed-88ee-ffc86e0cde67",
+                                  "threadId": "5e2b02d0-4ab0-11ed-88ee-ffc86e0cde67",
+                                  "size": 600
+                              }
+                          }
+                      },
+                      "0"
```
Comment (dab246) [inline test/jmap/email/set/set_email_method_test.dart:290]: it always starts at c0 if you don't set methodId

**Review by dab246 — COMMENTED** [review]
Sample: 
```dart
test('set email method and response parsing with header MDN', () async {
      final baseOption  = BaseOptions(method: 'POST');
      final dio = Dio(baseOption)
        ..options.baseUrl = 'http://domain.com/jmap';
      final dioAdapter = DioAdapter(dio: dio);
      dioAdapter.onPost(
          '',
          (server) => server.reply(200, {
            "sessionState": "2c9f1b12-b35a-43e6-9af2-0106fb53a943",
            "methodResponses": [[
              "Email/set",
              {
                "accountId":"3ce33c876a726662c627746eb9537a1d13c2338193ef27bd051a3ce5c0fe5b12",
                "oldState":"234c9ee0-0596-11ec-b153-2fef1ee78d9e",
                "newState":"234c9ee0-0596-11ec-b153-2fef1ee78d9e",
                "created": {
                  "aa1234": {
                    "id":"29a7f870-0596-11ec-b153-2fef1ee78d9e",
                    "blobId":"29a7f870-0596-11ec-b153-2fef1ee78d9e",
                    "threadId":"29a7f870-0596-11ec-b153-2fef1ee78d9e",
                    "size": 657
                  }
                }
              },
              "c0"
            ]]
          }),
          data: {
            'using': [
              'urn:ietf:params:jmap:mail',
              'urn:ietf:params:jmap:core'
            ],
            'methodCalls': [[
              'Email/set',
              {
                'accountId': '3ce33c876a726662c627746eb9537a1d13c2338193ef27bd051a3ce5c0fe5b12',
                'create': {
                  'aa1234': {
                    'id': 'ea12345',
                    'mailboxIds': {
                      'fe00a5c0-0584-11ec-b153-2fef1ee78d9e': true
                    },
                    'keywords': {
                      '\$seen': true
                    },
                    'subject': 'set email 3',
                    'sender': [{
                      'name: bob, email: bob@email'
                    }],
                    'from' : [{
                      'name': 'alice', 'email': 'alice@email'
                    }],
                    'to': [{
                      'name': 'dcu', 'email': 'dcu@email'
                    }],
                    'replyTo': [{
                      'name': 'bob', 'email': 'bob@email'
                    }],
                    'htmlBody': [{
                      'partId': 'a49d', 'type': 'text/html'
                    }],
                    'bodyValues': {
                      'a49d': {
                        'value': 'test html html',
                        'isEncodingProblem': false,
                        'isTruncated': false
                      }
                    },
                    'header:Disposition-Notification-To:asText': 'bob@email'
                  }
                }
              },
              'c0'
            ]
            ]
          },
          headers: {
            "accept": "application/json;jmapVersion=rfc-8621",
            "content-type": "application/json; charset=utf-8"
          }
      );

      final setEmailMethod = SetEmailMethod(AccountId(Id('3ce33c876a726662c627746eb9537a1d13c2338193ef27bd051a3ce5c0fe5b12')))
        ..addCreate(Id('aa1234'),
            Email(
                EmailId(Id('ea12345')),
                mailboxIds: {
                  MailboxId(Id('fe00a5c0-0584-11ec-b153-2fef1ee78d9e')): true
                },
                keywords: {
                  KeyWordIdentifier.emailSeen: true
                },
                replyTo: {EmailAddress('bob', 'bob@email')},
                from: {EmailAddress('alice', 'alice@email')},
                sender: {EmailAddress('bob', 'bob@email')},
                to: {EmailAddress('dcu', 'dcu@email')},
                subject: 'set email 3',
                htmlBody: {EmailBodyPart(partId: PartId('a49d'), type: MediaType.parse('text/html'))},
                bodyValues: {
                  PartId('a49d'): EmailBodyValue('test html html', false, false)
                },
                headerMdn: {IndividualHeaderIdentifier.headerMdn : 'bob@email'}
            )
        );

      final httpClient = HttpClient(dio);
      final requestBuilder = JmapRequestBuilder(httpClient, ProcessingInvocation());
      final setEmailInvocation = requestBuilder.invocation(setEmailMethod);
      final response = await (requestBuilder..usings(setEmailMethod.requiredCapabilities))
          .build()
          .execute();

      final setEmailResponse = response.parse<SetEmailResponse>(
          setEmailInvocation.methodCallId,
          SetEmailResponse.deserialize);

      expect(setEmailResponse!.created![Id('aa1234')], equals(expectedCreated));
    });

```

**Review by dab246 — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**